lib/kubeclient/common.rb

```diff
@@ -1,255 +1,299 @@
 require 'json'
 require 'rest-client'
 module Kubeclient
-  module Common
-    # Common methods
-    class Client
-      def handle_exception
-        yield
-      rescue RestClient::Exception => e
-        begin
-          json_error_msg = JSON.parse(e.response || '') || {}
-        rescue JSON::ParserError
-          json_error_msg = {}
-        end
-        err_message = json_error_msg['message'] || e.message
-        raise KubeException.new(e.http_code, err_message)
-      end
-
-      def handle_uri(uri, path)
-        @api_endpoint = (uri.is_a? URI) ? uri : URI.parse(uri)
-        @api_endpoint.path = path if @api_endpoint.path.empty?
-        @api_endpoint.path = @api_endpoint.path.chop \
-                           if @api_endpoint.path.end_with? '/'
+  # Common methods
+  module ClientMixin
+    attr_reader :api_endpoint
+    attr_reader :ssl_options
+    attr_reader :auth_options
+    attr_reader :headers
+
+    def initialize_client(
+      uri,
+      path,
+      version = nil,
+      ssl_options: {
+        client_cert: nil,
+        client_key: nil,
+        ca_file: nil,
+        verify_ssl: nil
```

Comment:
@moolitayer I think it's safer to set `verify_ssl` to `OpenSSL::SSL::VERIFY_PEER` here as well.